core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -634,6 +637,20 @@ private SqlNode getIdentifier(SqlBasicCall intervalOperand) {
     return intervalOperand.operand(0);
   }
 
+  private void unparseOther(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    switch (call.getOperator().getName()) {
+    case "DAYOFYEAR":
+      final SqlWriter.Frame frame = writer.startFunCall("EXTRACT");
+      writer.sep("DAYOFYEAR FROM");
+      call.operand(0).unparse(writer, 0, 0);
+      writer.endFunCall(frame);
+      break;
+    default:
+      super.unparseCall(writer, call, leftPrec, rightPrec);
+      break;
+    }
+  }
+
```

Comment:
ReltoSqlConverterUtils is a class where you can put the common code present in multiple dialects move the unparse method to that class and use that method here